src/coreclr/src/vm/class.h

```diff
@@ -372,63 +375,9 @@ class EEClassLayoutInfo
 #ifdef DACCESS_COMPILE
     friend class NativeImageDumper;
 #endif
-
-    private:
-        static void ParseFieldNativeTypes(
-            IMDInternalImport* pInternalImport,
-            const mdTypeDef cl, // cl of the NStruct being loaded
-            HENUMInternal* phEnumField, // enumerator for fields
-            const ULONG cTotalFields,
-            Module* pModule, // Module that defines the scope, loader and heap (for allocate FieldMarshalers)
-            ParseNativeTypeFlags nativeTypeFlags,
-            const SigTypeContext* pTypeContext, // Type parameters for NStruct being loaded
-            BOOL* fDisqualifyFromManagedSequential,
-            LayoutRawFieldInfo* pFieldInfoArrayOut, // caller-allocated array to fill in.  Needs room for cTotalFields+1 elements
-            EEClassLayoutInfo* pEEClassLayoutInfoOut, // caller-allocated structure to fill in.
-            ULONG* cInstanceFields // [out] number of instance fields
-#ifdef _DEBUG
-            ,
-            LPCUTF8 szNamespace,
-            LPCUTF8 szName
-#endif
-        );
-
-        static void SetOffsetsAndSortFields(
-            IMDInternalImport* pInternalImport,
-            const mdTypeDef cl,
-            LayoutRawFieldInfo* pFieldInfoArray, // An array of LayoutRawFieldInfos.
-            const ULONG cInstanceFields,
-            const BOOL fExplicitOffsets,
-            const UINT32 cbAdjustedParentLayoutNativeSize,
-            Module* pModule, // Module that defines the scope for the type-load
-            LayoutRawFieldInfo** pSortArrayOut // A caller-allocated array to fill in with pointers to elements in pFieldInfoArray in ascending order when sequential layout, and declaration order otherwise.
-        );
-
-        static void CalculateSizeAndFieldOffsets(
-            const UINT32 parentSize,
-            ULONG numInstanceFields,
-            BOOL fExplicitOffsets,
-            LayoutRawFieldInfo* const* pSortedFieldInfoArray, // An array of pointers to LayoutRawFieldInfo's in ascending order when sequential layout.
-            ULONG classSizeInMetadata,
-            BYTE packingSize,
-            BYTE parentAlignmentRequirement,
-            BOOL calculatingNativeLayout,
-            EEClassLayoutInfo* pEEClassLayoutInfoOut // A pointer to a caller-allocated EEClassLayoutInfo that we are filling in.
-        );
-
-        // size (in bytes) of fixed portion of NStruct.
-        UINT32      m_cbNativeSize;
         UINT32      m_cbManagedSize;
```

Comment:
Is this different from `GetNumInstanceFieldsBytes`?

Reply:
If `GetNumInstanceFieldsBytes` also accounts for `[StructLayout(..., Size = x)]`, then they should be identical.

Reply:
I believe these should be identical for blittable structs.

Reply:
The value for `GetNumInstanceFieldBytes` is actually calculated from the value stored here:

https://github.com/jkoritzinsky/runtime/blob/lazy-interop-data-generation/src/coreclr/src/vm/methodtablebuilder.cpp#L1780

It's the same in master

https://github.com/dotnet/runtime/blob/6691dfd1358248eafa641661b5d1e08ebcde2e51/src/coreclr/src/vm/methodtablebuilder.cpp#L1781-L1782

Reply:
It means we should not need to store it in `EEClassLayoutInfo`. `EEClassLayoutInfo` is field in EEClass and it has process lifetime. Do you agree?

(If this minor redundancy is hard to fix, it can wait for later.)

Reply:
Let's leave this till later. It's likely going to be somewhat annoying to untangle and this PR is already pretty large.